File: 20180712/Missing Number.py
class Solution(object):
    def missingNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        sum_nums = sum(nums)
        whole_n = len(nums)
        add_whole = (0 + whole_n) * (whole_n + 1) / 2
        return add_whole - sum_nums

File: 20180712/Missing Number.java
class Solution {
    public int missingNumber(int[] nums) {
        int whole_n = nums.length;
        int add_whole = (0 + whole_n) * (whole_n + 1) / 2;
        for(int i = 0; i < nums.length; i ++){
            add_whole -= nums[i];
        }
        return add_whole;
    }
}
